base/missing.jl

```diff
@@ -345,10 +354,19 @@ mapreduce_impl(f, op, A::SkipMissing, ifirst::Integer, ilast::Integer) =
         i == typemax(typeof(i)) && return Some(op(f(a1), f(a2)))
         i += 1
         v = op(f(a1), f(a2))
-        @simd for i = i:ilast
-            @inbounds ai = A[i]
-            if ai !== missing
-                v = op(v, f(ai))
```

Comment:
On master vectorizer shows the following to `sum(skipmissing(Vector{Union{Missing,Float64}}(randn(4096))))` :
```julia
LV: Checking a loop in "julia_mapreduce_impl_2411" from simdloop.jl:75 @[ missing.jl:348 ]
LV: Loop hints: force=? width=0 interleave=0
LV: Found a loop: L137
LV: Found an induction variable.
LV: Not vectorizing: Found an unidentified PHI   %value_phi30161 = phi double [ %117, %L137.lr.ph ], [ %value_phi33, %L137 ]
LV: Interleaving disabled by the pass manager
LV: Can't vectorize the instructions or CFG
LV: Not vectorizing: Cannot prove legality.
```

And the following for `sum(skipmissing(Vector{Union{Missing,Int}}(rand(-1:1,4096))))`:
```julia
LV: Checking a loop in "julia_mapreduce_impl_2505" from simdloop.jl:75 @[ missing.jl:348 ]
LV: Loop hints: force=? width=0 interleave=0
LV: Found a loop: L137
LV: Found an induction variable.
LV: We can vectorize this loop!
```

Reply:
Weird that `LV: Not vectorizing: Found an unidentified PHI   %value_phi30161 = phi double [ %117, %L137.lr.ph ], [ %value_phi33, %L137 ]` is incomplete there is only one: `Found an unidentified PHI` message and that is longer.

Reply:
I tried the following:
```julia
function f(a)
    r = zero(eltype(a))
    @inbounds @simd for i in eachindex(a)
       if a[i] > 0
           r += a[i]
       end
   end
   r
end
function g(a)
   r = zero(eltype(a))
   @inbounds @simd for i in eachindex(a)
       r += a[i] > 0 ? a[i] : 0
   end
   r
end
```
`g(randn(4096))` is vectorlized by LLVM while `f(randn(4096))` not. I tried rewrite `f(a)` with `c`, the output of Clang shows some simd IR (Although the simd width is only 2). Maybe related with #31862?